pkg/member/member_control.go

```diff
@@ -319,17 +318,13 @@ func (m *memberControl) IsLearnerPresent(ctx context.Context) (bool, error) {
 // IsClusterScaledUp determines whether a etcd cluster is getting scale-up or not and returns a boolean
 func (m *memberControl) IsClusterScaledUp(ctx context.Context, clientSet client.Client) (bool, error) {
 	m.logger.Info("Checking whether etcd cluster is marked for scale-up")
-	curSts := &appsv1.StatefulSet{}
-	err := clientSet.Get(ctx, client.ObjectKey{
-		Namespace: m.podNamespace,
-		Name:      m.podName,
-	}, curSts)
+	etcdsts, err := miscellaneous.GetStatefulSet(ctx, clientSet, m.podNamespace, m.podName)
 	if err != nil {
-		m.logger.Errorf("unable to fetch etcd sts: %v", err)
-	}
-
-	if err == nil && curSts != nil && miscellaneous.IsAnnotationPresent(curSts, miscellaneous.ScaledToMultiNodeAnnotationKey) {
-		return true, nil
+		m.logger.Errorf("unable to fetch etcd statefulset: %v", err)
+	} else {
+		if etcdsts != nil && miscellaneous.IsAnnotationPresent(etcdsts, miscellaneous.ScaledToMultiNodeAnnotationKey) {
```

Comment:
can `etcdsts` ever be nil?

Reply:
It is just a safety check.

Reply:
ok. It is not really required. If it is nil then we should let it fail as that would be quite weird and totally unexpected. If we do this in one place then we have to make sure that we do it everywhere else as well. This is called `defensive programming`

Reply:
removed.